README.rst

```diff
@@ -6,126 +6,122 @@ Taisei
+source archives, as they don't contain the necessary submodules for compiling
+the game. Make sure you either download the release tarball, or use `git clone`.
 
-    cd /path/to/taisei/source
-    mkdir build
-    cd build
-    meson --prefix=$yourprefix ..
-    ninja
-    ninja install
+You can obtain the source code of the stable releases from the
+`Releases <https://github.com/taisei-project/taisei/releases>`__ page on
+GitHub, alongside the binaries.
 
-This will install game data to ``$prefix/share/taisei/`` and build this
-path *statically* into the executable. This might be a package
-maintainer’s choice. Alternatively you may want to add
-``-Dinstall_relative=true`` to get a relative structure like
+You can also grab the latest code from git using the following commands:
 
 ::
 
-    $prefix/taisei
-    $prefix/data/
+    git clone --recurse-submodules https://github.com/taisei-project/taisei
 
-``install_relative`` is always set when building for Windows.
+The ``--recursive-submodules`` option is important as the submodules are
+necessary to compile the project.
 
-The OpenGL ES 3.0 backend is not built by default. To enable it, do:
+If you want to pull them in manually, be sure to run the following:
 
 ::
 
-    meson configure -Dr_gles30=true -Dshader_transpiler=true
+    cd taisei
+    git submodule init
+    git submodule update
+
+You should also run ``git submodule update`` whenever you pull in
+new code, checkout another branch, etc. The ``pull`` and ``checkout`` helper
+scripts can do that for you automatically.
 
-See `here <doc/ENVIRON.rst>`__ for information on how to activate it.
-Alternatively, do this to make GLES 3.0 the default backend:
+For more in-depth development and compiling instructions, see our
+`Development Guide <doc/DEVELOPMENT.rst>`__.
 
-::
+Basic Dependencies
+^^^^^^^^^^^^^^^^^^
 
-    meson configure -Dr_default=gles30
+-  OpenGL >= 3.3, or OpenGL ES >= 3.0, or OpenGL ES >= 2.0 (with extensions)
+-  SDL2 >= 2.0.10
+-  Python >= 3.6
+-  meson >= 0.56.0
+-  freetype2
+-  libpng >= 1.5.0
+-  libwebpdecoder >= 0.5 or libwebp >= 0.5
+-  libzip >= 1.5.0 (>= 1.7.0 recommended)
+-  libzstd >= 1.4.0
+-  opusfile
+-  zlib
 
-The OpenGL ES 2.0 backend can be enabled similarly, using ``gles20`` instead of
-``gles30``. However, it requires a few extensions to function correctly, most
-notably:
+Optional:
 
-- ``OES_depth_texture`` or ``GL_ANGLE_depth_texture``
-- ``OES_standard_derivatives``
-- ``OES_vertex_array_object``
-- ``EXT_frag_depth``
-- ``EXT_instanced_arrays`` or ``ANGLE_instanced_arrays`` or
-  ``NV_instanced_arrays``
+-  OpenSSL (for a better SHA-256 implementation; used in shader cache)
+-  SPIRV-Cross >= 2019-03-22 (for OpenGL ES backends)
+-  libshaderc (for OpenGL ES backends)
+-  ANGLE (for ANGLE support, required for Windows)
+-  GameMode headers (Linux only; for automatic `GameMode
+   <https://github.com/FeralInteractive/gamemode>`__ integration)
```

Comment:
Could of things here

* I don't like the merging of runtime dependencies with build-time dependencies. The distinction is pretty important particularly in a cross-compilation environment.
* cglm is now missing (runtime dependency)
* python-zstandard is also missing (build dependency)
* ANGLE is not "required" on windows. For the purposes of hacking on the codebase, you probably won't even need it, unless you're stuck with an Intel iGPU.
* I think we can just remove mentions of OpenGL ES 2.0, and simply state that GL3.3+ or ES3.0+ is required. The gles20 backend is unmaintained and doesn't really work; at best it'll only work on ES3-capable hardware.